config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,37 @@
+if defined?(Rails::Server) && !Rails.env.test?
+  require 'rack/attack'
+
+  # The `rack/attack/railtie.rb` creates a `initializer` call, which is loaded
+  # too late because of the `:require => false` in our Gemfile, so this needs
+  # to be done manually.
+  Rails.application.middleware.use(Rack::Attack)
```

Comment:
Note, the `require => false` is not entirely required if it's causing us technical debt.  From looking at their [railtie](https://github.com/rack/rack-attack/blob/master/lib/rack/attack/railtie.rb#L7), it looks very simple and so your workaround looks great, but if they add any complexity or dependence on their code loading with the rails initialization, we should certainly reconsider allow it to be required.

Reply:
Yeah, I am down for either on this route.  I don't think `rack-attack` is a particularly large dependency to hold in memory, so it probably wouldn't hurt having it loaded, even when it isn't needed.

But at the same time, this isn't a very large hack, so I don't have to big of an issue with it being here if we are continuing a trend of trying to keep workers as lean as possible.

* * *

With that said, it might possibly make sense to add a test or two around this, so if how we do this does change, we can confirm the limiting is still working as expected and not being configured with out being activated.

Reply:
Yeah, exactly, it's no big deal either way right now.